### PR TITLE
feat(MainActivity): provide "hardwareType" and "Model" from Build as "android.prop"

### PR DESCRIPTION
**Pull request by hasezoey**
This PR provides `Build.HARDWARE` (`ro.product.hardwareType`) and `Build.MODEL`(`ro.product.model`) in lua for a PR for the `frontend` to detect Tolino

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/android-luajit-launcher/381)

### CODE REVIEW

File: app/src/main/java/org/koreader/launcher/LuaInterface.kt [hunk 1]
```diff
@@ -43,6 +43,8 @@ interface LuaInterface {
     fun getScreenWidth(): Int
     fun getStatusBarHeight(): Int
     fun getVersion(): String
+    fun getHardwareType(): String
+    fun getModel(): String
```
Comment (Frenzie) [inline app/src/main/java/org/koreader/launcher/LuaInterface.kt:47]: I think they're supposed to be more alphabetical?

**Review by Frenzie — APPROVED** [review]
lgtm give or take alphabetization nits but @pazos has the final word regardless

File: app/src/main/java/org/koreader/launcher/LuaInterface.kt (hunk 1 above)
Comment (hasezoey) [reply app/src/main/java/org/koreader/launcher/LuaInterface.kt:47]: an change, but i had added it below `getVersion` to have it in a similar place

Comment (hasezoey) [reply app/src/main/java/org/koreader/launcher/LuaInterface.kt]: fixed with e77d0b848b86ebec7ea009acb7eed4fa0b635eed

**Comment by pazos** [comment]
Hey @hasezoey, `android.product` is enough to identify a given tolino model.

If you want to export `Build` things to lua keep in mind we're talking about java Strings, so please double check they're not nullable. But that's already done in https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/device/DeviceInfo.kt#L168-L173
with https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/device/DeviceInfo.kt#L528-L534

So please repurpose those instead.

**Comment by hasezoey** [comment]
updated the 2 functions to use `DeviceInfo` instead

**Comment by pazos** [comment]
superseeded by #383

**Comment by hasezoey** [comment]
closing in favor of #383

**Pull request closed without merging**